WordPress/gutenberg
# Navigation: Select dropdown encapsulation and further consolidation.

## Description
This PR builds on https://github.com/WordPress/gutenberg/pull/38179 to further consolidate and normalize all code paths around the "Select menu" dropdown that is shown in:

* placeholder
* block toolbar

This PR moves all permissions checking and data fetching within the `<NavigationMenuSelector>` component and then utilises that single component in both instances (as above).

This ensures all logic is the same across all areas where this UI is exposed.

Also addresses part of https://github.com/WordPress/gutenberg/issues/37190.

## Testing Instructions

The aims is that nothing changes for the block UI. It should be "as was".

1. New post.
2. Add a Nav block.
3. See "Select menu" - click to reveal dropdown. Test each of the options underneath (existing, existing classic).
4. Create another Nav block. Click "Create empty".
5. In block toolbar click "Select menu". Test all of the options still work (existing, existing classic, manage, new).


## Screenshots 

## Types of changes





## Checklist:
